Add accountID to GC message

KillianG opened the pull request:
Issue: BB-503

Commit:
Add accountID to GC message

Issue: BB-503

bert-e commented:
# Hello killiang,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-503 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.6.37`

* `8.7.0`


Please check the `Fix Version/s` of BB-503, or the target
branch of this pull request.

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

Commit:
Fix error when bucket not existing

When a bucket doesnt exist we want the offset to be commited because there is no need to retry that again as the object as already been deleted somehow

Issue: BB-504

Commit:
Add test for non existing bucket

GC tasks

Issue: BB-504

francoisferrand commented on extensions/lifecycle/tasks/LifecycleUpdateExpirationTask.js:
```diff
@@ -126,7 +126,8 @@ class LifecycleUpdateExpirationTask extends BackbeatTask {
               .setAttribute('serviceName', 'lifecycle-transition')
-              .setAttribute('target.locations', locations);
+              .setAttribute('target.locations', locations)
+              .setAttribute('target.accountId', accountId);
```
should also (try to) pass the `ownerId`, for good measure (like in the other cases)

francoisferrand commented on tests/unit/gc/GarbageCollectorTask.spec.js:
```diff
@@ -199,4 +199,33 @@ describe('GarbageCollectorTask', () => {
+              .setAttribute('target.key', version)
+              .setAttribute('target.version', key)
+              .setAttribute('target.accountId', accountId)
+              .setAttribute('target.owner', owner);
```
should it not be `target.ownerId` ?

Commit:
Adding owner id to garbage collector message

Issue: BB-503

Commit:
Change owner to ownerId

Issue: BB-503

francoisferrand commented on extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js:
```diff
@@ -36,7 +37,7 @@ class LifecycleColdStatusArchiveTask extends LifecycleUpdateTransitionTask {
               .setAttribute('target.version', version)
               .setAttribute('target.accountId', accountId)
-              .setAttribute('target.owner', owner);
+              .setAttribute('target.ownerId', ownerId);
```
this is a partial fix only.

Looking at backbeat (lifecycle) overall, there are _lots_ places where we set `target.owner` field : so either we should change them all, or change GC to extract the `owner` field from message.

Commit:
Extract owner instead of ownerId

Issue: BB-503

francoisferrand commented on extensions/gc/tasks/GarbageCollectorTask.js:
```diff
@@ -23,16 +23,16 @@ class GarbageCollectorTask extends BackbeatTask {
     // TODO BB-367: replace once replication services uses assume role
     _getAccountId(entry, log, cb) {
-        const { accountId, ownerId } = entry.getAttribute('target');
+        const { accountId, owner } = entry.getAttribute('target');
```
also needs to be changed in https://github.com/scality/backbeat/blob/c790d6de0e52b1ba701dd1fe7103362fb97ed093/extensions/replication/tasks/UpdateReplicationStatus.js#L279

KillianG commented:
/approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-503-missing-account-id-in-gc-message` into
`development/8.6`
* [w/8.7/bugfix/BB-503-missing-account-id-in-gc-message](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-503-missing-account-id-in-gc-message) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`



Please check the status of the associated issue BB-503.

Goodbye killiang.
